app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ReserveView.java

```diff
@@ -39,8 +41,40 @@ public void setEndDateTextView(String date) {
 
     @Override
     public void returnToMainActivity() {
-        Intent intent = new Intent(getActivity(), MainActivity.class);
-        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-        getActivity().startActivity(intent);
+        getActivity().finish();
+    }
+
+    @Override
+    public int getParkingLot() {
+        String lot = binding.editTextNewReservationActivityLotNumber.getText().toString();
+        if (lot.isEmpty())
+            return -1; //-1 used in model to check if user has set a parking lot
+        else return Integer.parseInt(lot);
+    }
+
+    @Override
+    public String getUserPassword() {
+        return binding.editTextNewReservationActivityPassword.getText().toString();
+    }
+
+    @Override
+    public void showComprobationToast(ReserveComprobation comprobation) {
+        switch (comprobation) {
```

Comment:
this switch should not be in the view, should be in the presenter, and remember to check that the getActivity is not null before using it

Reply:
done